client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -4,23 +4,82 @@ import { deleteTea, getTeas } from "../../../actions/teaActions";
 import TeaList from "./TeaList";
 
 export class TeaListContainer extends React.Component {
+  state = {
+    sortedIDs: [],
+    memoizedIDs: {},
+    sortColumn: "",
+    sortOrder: ""
+  };
+
   handleDeleteClick = tea => {
     this.props.handleDelete(tea);
   };
 
+  handleSortClick = (columnName, sortOrder) => {
+    const list = this.props.teas.teaIDs;
+    let currentState = this.state.memoizedIDs;
+    let newSortOrder;
+
+    if (this.state.memoizedIDs[columnName]) {
+      newSortOrder = this.state.memoizedIDs[columnName][sortOrder];
+    } else {
+      let mapped = list.map((el, i) => {
+        return {
+          index: i,
+          value: this.props.teas.allTeas[el][columnName]
+        };
+      });
+
+      mapped.sort((a, b) => {
+        if (a.value > b.value) {
+          return 1;
+        }
+        if (a.value < b.value) {
+          return -1;
+        }
+        return 0;
+      });
+
+      newSortOrder = mapped.map(el => list[el.index]);
```

Comment:
If you use the suggestion in the comment above, this becomes `newSortOrder = mapped.map(el => el.id);` Up to you, but I think that logic is a bit easier to follow.